lib/text_object.js

```diff
@@ -1,130 +1,68 @@
-class TextNode extends SizeNode{
-    /**
-     * 
-     * @param {string} name
-     * @param {string} text 
-     * @param {boolean} message_control 
-     * @param {Object.<string, any>|undefined} style 
-     */
-    constructor(name, text, message_control=true, style){
-        super(name, 10, 10, message_control, false)
+class TextNode extends SizeNode {
+    constructor(name, str, message_control=true){
+        super(name, 1, 1, message_control, false)
         
-        this.text = text
-        this.element = createP(text);
+        this._text = str;
+        this._font = null;
+        this._fontSize = 12;
+        this._fontStyle = NORMAL;
+        this._align = LEFT;
+        this._lineHeight = this.fontSize;
+        this._color = [0, 0, 0, 255];
 
-        if (style != undefined) {
-            for (var key in style) {
-                this.element.style(key, style[key]);
-                // save styles in state dict
-                this.state[key] = style[key]
+        this.drawSelf = function() {
+            if(this._font != null) {
+                textFont(this._font);
             }
-        }
+            textSize(this._fontSize);
+            textStyle(this._fontStyle);
+            textAlign(this._align);
+            textLeading(this._lineHeight);
+            fill(...this._color);
 
-        this.element.style("overflow", "hidden")
-        //this.element.style("left", "1%")
-        this.element.style("top", "1%")
-        this.element.style("right", "1%")
-        //this.element.style("bottom", "1%")
-        this.element.style("text-align", "center")
-        //this.element.style("margin", "1%") 
-        this.element.style("height","fit-content")
-        this.element.style("width","fit-content")
-        /*
-        const halfWidthMatches = (text.match(/[\u0000-\u00ff]/g)) //Half-width characters
-        const chineseMatches = (text.match(/[\u4e00-\u9fa5]/g)) //Chinese characters
-        const fullWidthMatches = (text.match(/[\uff00-\uffff]/g)) //full width chracters
-        console.log(halfWidthMatches, chineseMatches, fullWidthMatches)
-        const numHalf = halfWidthMatches ? halfWidthMatches.length : 0
-        const numChinese = chineseMatches ? chineseMatches.length : 0
-        const numFull = fullWidthMatches ? fullWidthMatches.length : 0
-        console.log(numHalf, numChinese, numFull)
-        const numEM = Math.ceil(numHalf / 2) + numChinese + numFull
-        console.log(numEM)
-        this.element.style("width", numEM +"em")
-        */
+            text(this._text, ...this.translation);
```

Comment:
You should ignore the transform here .

The transformation tracker in the rendering pipeline takes cares of the accumulated transformations of each `Node` instance (including 1. translation 2. rotation 3. scaling  4. alpha  5. colortint ) in the scene graph. Hence one only need to draw the object at origin in `Node.drawSelf()` function. 

Also, this is the cause of text position mismatch in `t_text_scaling()` of `test/text_node_test`
![image](https://user-images.githubusercontent.com/32272431/200123997-f489ceff-1472-4a36-b167-f7cea659f865.png)